cmd/start.go

```diff
@@ -91,3 +116,44 @@ func trapSignal(done func()) {
 	// call the cleanup func
 	done()
 }
+
+// UpdateClientsFromChains takes src, dst chains and update clients based on expiry time
+func UpdateClientsFromChains(src, dst *relayer.Chain) (sleepTime time.Duration, err error) {
+	var srcTimeExpiry, dstTimeExpiry time.Duration
+
+	thresholdTime := viper.GetDuration(flagThresholdTime)
+
+	eg := new(errgroup.Group)
+	eg.Go(func() error {
+		srcTimeExpiry, err = relayer.GetClientAndUpdate(src, dst, thresholdTime)
+		return err
+	})
+	eg.Go(func() error {
+		dstTimeExpiry, err = relayer.GetClientAndUpdate(dst, src, thresholdTime)
+		return err
+	})
+	if err := eg.Wait(); err != nil {
+		return 0, err
+	}
+
+	if srcTimeExpiry > 0 {
```

Comment:
This can be massively simplified by just using math.Min.

If either `srcTimeExpiry` or `dstTimeExpiry` is less than 0 just return an error.